sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -52,585 +21,237 @@ const DISBURSEMENT_SCHEDULES_UPDATE_CHUNK_SIZE = 1000;
+          },
+          offering: {
+            offeringIntensity,
+          },
+        },
+        student: {
+          studentRestrictions: [{ isActive: IsNull() }, { isActive: true }],
+        },
+      },
+      order: {
+        currentAssessment: {
+          disbursementSchedules: {
+            disbursementDate: "ASC",
+            createdAt: "ASC",
+          },
+        },
+      },
     });
-    // Step 6 - Save all changes for the disbursements schedules and awards.
-    await entityManager
-      .getRepository(DisbursementSchedule)
-      .save(disbursements, {
-        chunk: DISBURSEMENT_SCHEDULES_UPDATE_CHUNK_SIZE,
+    // Convert the application records to be returned as disbursements to allow
+    // easier processing along the calculation steps.
+    const eligibleDisbursements =
+      eligibleApplications.flatMap<EligibleECertDisbursement>((application) => {
+        return application.currentAssessment.disbursementSchedules.map(
+          (disbursement) => {
+            return {
+              studentId: application.student.id,
+              assessmentId: application.currentAssessment.id,
+              applicationId: application.id,
+              // Convert the nested restriction in StudentRestriction to a simple object.
+              activeRestrictions: application.student.studentRestrictions.map(
+                (studentRestriction) => {
+                  return {
+                    id: studentRestriction.restriction.id,
+                    actions: studentRestriction.restriction.actionType,
+                  };
+                },
+              ),
+              hasValidSIN: application.student.sinValidation.isValidSIN,
+              disbursement,
+              offering: application.currentAssessment.offering,
+              maxLifetimeBCLoanAmount:
+                application.programYear.maxLifetimeBCLoanAmount,
+            };
+          },
+        );
       });
-    return disbursements;
+    return eligibleDisbursements;
   }
 
   /**
    * Get all records that must be part of the e-Cert files and that were not sent yet.
-   * Criteria to be a valid disbursement to be sent.
-   * - Not sent yet;
-   * - Disbursement date in the past or in the near future (defined by DISBURSEMENT_FILE_GENERATION_ANTICIPATION_DAYS);
-   * - Student had the SIN validated by the CRA;
-   * - Student has a valid signed MSFAA and not cancelled;
-   * - No restrictions in the student account that prevents the disbursement;
-   * - Application status must be 'Completed';
-   * - Confirmation of enrollment(COE) must be 'Completed'.
-   * - Disbursement schedule on pending status.
    * @param offeringIntensity disbursement offering intensity.
    * @param entityManager used to execute the commands in the same transaction.
    * @returns records that must be part of the e-Cert.
    */
-  private async getECertInformationToBeSent(
+  async getReadyToSendDisbursements(
```

Comment:
Refactored from `prepareDisbursementsForECertGeneration` removing unnecessary fields that are not needed for the e-Cert generation.